### PR TITLE
DO NOT MERGE YET: testing CI

### PR DESCRIPTION
**Pull request by matejvasek**

### CODE REVIEW

**Comment by matejvasek** [comment]
@lkingland trying to run CI on multiple OSes.

**Comment by matejvasek** [comment]
@lkingland  there are few tests failing because of file mods (access perms.); I am not sure how we could fix this.

File: function_unit_test.go [hunk 1]
```diff
@@ -23,50 +23,50 @@ func TestPathToDomain(t *testing.T) {
 		{"", noLimit, ""},
 
 		// trailing slashes ignored
-		{"/home/user/example.com/admin/", noLimit, "admin.example.com"},
+		{"/home" + string(os.PathSeparator) + "user" + string(os.PathSeparator) + "example.com" + string(os.PathSeparator) + "admin/", noLimit, "admin.example.com"},
```
Comment (lkingland) [inline function_unit_test.go:26]: This might be cleaner as `filepath.Join(...)`

File: .github/workflows/pull_requests.yaml [hunk 2]
```diff
@@ -5,7 +5,10 @@ on: [pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
+    runs-on: ${{ matrix.os }}
```
Comment (lkingland) [inline .github/workflows/pull_requests.yaml:11]: I love that we are dipping our toes in here, but I would caution that we don't spend too much time on cross-compatibility until we're past v1, or we risk spending a lot of time in maintenance of platforms not in active use.

**Comment by lkingland** [comment]
> @lkingland there are few tests failing because of file mods (access perms.); I am not sure how we could fix this.

Sounds like a bug in pkger to me.  I created a new issue #57

File: .github/workflows/pull_requests.yaml (hunk 2 above)
Comment (lance) [reply .github/workflows/pull_requests.yaml:11]: Although I would say that OSX and Linux are must-have...

**Comment by lance** [comment]
> @lkingland there are few tests failing because of file mods (access perms.); I am not sure how we could fix this.

I think [this test](https://github.com/boson-project/faas/blob/develop/function_unit_test.go#L27-L70) could be changed to be OS agnostic. And [this test](https://github.com/boson-project/faas/blob/develop/embedded/initializer_test.go#L122-L124) could check `runtime.GOOS` for `"windows"` and have a special test just for that platform.

**Comment by lance** [comment]
@matejvasek this PR is now pointing to the incorrect main branch (`develop`) and CI has changed so much that I think this should be closed and reconsidered for a future PR.

**Pull request closed without merging**